### ISSUE TITLE
Rider configuration

### ISSUE THREAD

**Issue by atzimler** [JetBrains/resharper-cyclomatic-complexity#18]
Hi,

I haven't been able to find the configuration for the Rider version of the plugin. Is there any? I've tried to look for Cyclomatic or Complexity words in the preferences pane of Rider.

Thanks for the help,
Cheers,
Attila

**Comment by matkoch** [comment]
@atzimler there is currently no options page for Rider, because the existing options are based on XAML, which won't work with the JVM. So it's going to be reimplemented in some way. I will take a look eventually.

**Comment by panpawel88** [comment]
Hello,

I'm looking for the same option. I was wondering, is there any way to change plugin configuration outside Rider IDE?

Pawel

**Comment by atzimler** [comment]
Yes, it would be really nice at least to be able to adjust the CCM level. I would like to set it to 5 instead of the default 20 for warning and probably that's all that I would like to configure on the plugin for the rest of the IDE's lifetime :)

**Comment by matkoch** [comment]
I’m going to look into this over the holidays. Please forgive me. Just FYI, you can still manually set this using dotsettings files, just in case someone doesn’t know about this.

**Comment by atzimler** [comment]
The DotSettings configuration would be more than enough for me for now. Could you give some direction on finding documentation regarding to that? I've tried to export my settings, I don't see the values in it. I've tried to look into the DotSettings.user in the project directory, that was also not helping, and wasn't able to easily find the source code part reading in the configuration from the plugin source code.

**Comment by matkoch** [comment]
It’s not put into dotsettings files unless it is different from default. The easiest way is to do this via ReSharper. In that case you don’t even need to look into the dotsettings files.

**Comment by matkoch** [comment]
And here is some example configuration for CSharp as language: https://github.com/nuke-build/nuke/blob/develop/source/Inspections.DotSettings#L338

**Comment by atzimler** [comment]
With version 2018.3 of Rider and the process of manually adding the suggested line above into the x.sln.DotSettings.user file, restarting Rider, reloading the project, the default level is still 20. The project is a C# project, so I would expect to change it to 7 from 20 with the given line.

**Comment by matkoch** [comment]
You don't need to reload the IDE. File changes should have immediate effect (I would probably at least still cut and copy the code to  bes ure). I rather suspect that the dotsettings file is not picked up. To validate that you're using the right one, go to *Settings*, then lower left side *Manage layers*, chose the user layer there, change a code style setting or something, and check if it actually does.

**Comment by atzimler** [comment]
My expectation was also not to need to restart the IDE, just wanted to be sure, after just reloading the solution it was not working. I don't have the mentioned menu entries, probably because I'm using the macOS version. Until know I was expecting this to work similarly because of the DotSettings file. Before I tried the .user file in the solution, I was also exporting the global settings, making the change in the Export.GlobalSettingsStorage.<xxx>.DotSettings rezipping the files and reimporting my settings, but that also did not work.

**Comment by matkoch** [comment]
Well I'm on Mac too  :)  Cmd+, then "Manage Layers" in the lower  left.

**Comment by atzimler** [comment]
Sorry, for the menu, I was looking into the list of configurations, it's early morning :D (in Australia). Changing the tab spacing actually updates the expected DotSettings file, when saving the configuration into personal settings on the Save button, so I'm guessing it was the right DotSettings file. On a side note, saving the settings actually removed the manually added line.

**Comment by matkoch** [comment]
Tested here with https://github.com/matkoch/TestLinker

![2018-12-18_21-31-04](https://user-images.githubusercontent.com/5005566/50181379-80fbfe80-030c-11e9-8de7-fef43d07011d.gif)

Does the plugin appear in the plugin list?

**Comment by atzimler** [comment]
Yes, it is. The difference between your screen and my trial is that I've tried to add the line at the end of the settings file. It seems that for some reason it does not work there. If I add it after the first line in the settings it does work.

This configuration is working:
```
<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
	<s:Int64 x:Key="/Default/CodeInspection/CyclomaticComplexityAnalysis/Thresholds/=CSHARP/@EntryIndexedValue">5</s:Int64>
	<s:Boolean x:Key="/Default/CodeStyle/Naming/CSharpAutoNaming/IsNotificationDisabled/@EntryValue">True</s:Boolean>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=TypesAndNamespaces/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=StaticReadonly/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=PrivateInstanceFields/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="_" Suffix="" Style="aaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=Property/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=Method/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=Parameters/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="aaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=Interfaces/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="I" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=TypeParameters/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="T" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=Locals/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="aaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=EnumMember/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=PrivateStaticReadonly/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=Event/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=PrivateConstants/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=PrivateStaticFields/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="_" Suffix="" Style="aaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=Constants/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:Boolean x:Key="/Default/CodeStyle/Naming/CSharpAutoNaming/IsNamingAutoDetectionCompleted/@EntryValue">True</s:Boolean></wpf:ResourceDictionary>
```

This one doesn't:
```
<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
	<s:Boolean x:Key="/Default/CodeStyle/Naming/CSharpAutoNaming/IsNotificationDisabled/@EntryValue">True</s:Boolean>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=TypesAndNamespaces/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=StaticReadonly/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=PrivateInstanceFields/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="_" Suffix="" Style="aaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=Property/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=Method/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=Parameters/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="aaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=Interfaces/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="I" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=TypeParameters/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="T" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=Locals/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="aaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=EnumMember/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=PrivateStaticReadonly/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=Event/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=PrivateConstants/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=PrivateStaticFields/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="_" Suffix="" Style="aaBb" /&gt;</s:String>
	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/AutoDetectedNamingRules/=Constants/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="AaBb" /&gt;</s:String>
	<s:Boolean x:Key="/Default/CodeStyle/Naming/CSharpAutoNaming/IsNamingAutoDetectionCompleted/@EntryValue">True</s:Boolean></wpf:ResourceDictionary>
	<s:Int64 x:Key="/Default/CodeInspection/CyclomaticComplexityAnalysis/Thresholds/=CSHARP/@EntryIndexedValue">5</s:Int64>
```

**Comment by matkoch** [comment]
Yes, because the second is malformed XML :) Note the sneaky `</wpf:ResourceDictionary>`  at the end.

**Comment by atzimler** [comment]
AAAA. Why is it exported in a formatted, but badly formatted XML way? :D I was not even seeing the closing tag in Rider (highlight does not work automatically for this file), not just in the console text editor.

At least, I can configure it now, the settings export import trick with the corrected file still does not seem to work, but for now I'm really happy with the result. And going to drink a coffee after overlooking XML syntax and a button :D

**Comment by matkoch** [comment]
This is now fixed with 2018.3.1
